dashboard/src/features/shared/components/BaseNew.jsx

```diff
@@ -6,7 +6,11 @@ export const mapStateToProps = ( type ) => ( /* state */ ) => ({
 })
 
 export const mapDispatchToProps = (type) => (dispatch) => ({
-  submitForm: (data) => dispatch(actions[type].submitForm(data)),
+  submitForm: (data) => {
+    return dispatch(actions[type].submitForm(data)).then(() => {
+      dispatch(actions.tutorial.updateTutorial(data, type))
```

Comment:
This function should return `data` so that the calling function can chain more `.then`s onto the result if desired.